api/src/main/java/org/openmrs/module/fhir/api/util/FHIRPatientUtil.java

```diff
@@ -160,4 +169,120 @@ public static Patient generatePatient(org.openmrs.Patient omrsPatient) {
 		FHIRUtils.validate(patient);
 		return patient;
 	}
+	
+	public static org.openmrs.Patient generateOmrsPatient(Patient patient) {
+		org.openmrs.Patient omrsPatient=new org.openmrs.Patient();
+
+		List<IdentifierDt> fhirIdList = patient.getIdentifier();
+		IdentifierDt iddt = (IdentifierDt) fhirIdList.get(0);
+		PatientIdentifier idnt = new PatientIdentifier();
+		idnt.setIdentifier(iddt.getValue());
+		Location loc = Context.getLocationService().getLocationByUuid("8d6c993e-c2cc-11de-8d13-0010c6dffd0f");
+		idnt.setLocation(loc);
```

Comment:
@harsha89  @surangak I double checked this behaviour. you can create a identifier type which is not using locations. But whn you create a patient through API you have to set a location anyway. But when you read that new patient in UI, in the identifier field, you will not see any location resource. 
http://pastebin.com/zZzcy1Zf

error log